lib/rules/no-curly-component-invocation.js

```diff
@@ -66,6 +83,38 @@ module.exports = class NoCurlyComponentInvocation extends Rule {
     return `You are using the component {{${name}}} with curly component syntax. You should use <${angleBracketName}> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. \`'no-curly-component-invocation': { allow: ['${name}'] }\`.`;
   }
 
+  fixForBlockNode(node) {
+    if (node.program.type !== 'Block') {
+      return;
+    }
+    if (node.program.chained) {
+      return;
+    }
+    if (node.path.type !== 'PathExpression') {
+      return;
+    }
+    if (node.path.this === true) {
+      return;
+    }
+    if (node.path.data === true) {
+      return;
+    }
```

Comment:
Why do we short circuit in these two scenarios? Seems like we could properly transform both of these:


```hbs
{{#this.foo hi="wave"}}
{{/this.foo}}

{{#@foo lol="huzzah"}}
{{/@foo}}
```

Right? Maybe these guards were used when also fixing plain mustaches?

Reply:
@rwjblue good catch!

is it valid case:

```hbs
{{#let (component "foo") as |my-component|}}<my-component></my-component>{{/let}}
```

?

Reply:
Yes, this should work properly (though I _generally_ don't see folks use `-` in block params).